spec/controllers/state_file/questions/income_review_controller_spec.rb

```diff
@@ -100,6 +104,120 @@
+        let(:intake) { create(:state_file_az_intake) }
+        let!(:w2_1) { create(:state_file_w2, state_file_intake: intake) }
+        let!(:w2_2) { create(:state_file_w2, state_file_intake: intake) }
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when no w2s" do
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when only one w2 and box14_ui_wf_swf is not present" do
+        let!(:state_file_w2) { create(:state_file_w2, state_file_intake: intake, box14_fli: 145.26) }
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when only one w2 and fli is not present" do
+        let!(:state_file_w2) { create(:state_file_w2, state_file_intake: intake, box14_ui_wf_swf: 179.78) }
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when there are two w2s where box14_ui_wf_swf is not present but they have different filers" do
+        let(:intake) { create(:state_file_nj_intake, :married_filing_jointly) }
+        let(:primary_ssn_from_fixture) { intake.primary.ssn }
+        let(:spouse_ssn_from_fixture) { intake.spouse.ssn }
+        let!(:w2_1) { create(:state_file_w2, state_file_intake: intake, employee_ssn: primary_ssn_from_fixture, box14_fli: 145.26) }
+        let!(:w2_2) { create(:state_file_w2, state_file_intake: intake, employee_ssn: spouse_ssn_from_fixture, box14_fli: 145.26) }
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when there are two w2s where fli is not present but they have different filers" do
+        let(:intake) { create(:state_file_nj_intake, :married_filing_jointly) }
+        let(:primary_ssn_from_fixture) { intake.primary.ssn }
+        let(:spouse_ssn_from_fixture) { intake.spouse.ssn }
+        let!(:w2_1) { create(:state_file_w2, state_file_intake: intake, employee_ssn: primary_ssn_from_fixture, box14_ui_wf_swf: 179.78) }
+        let!(:w2_2) { create(:state_file_w2, state_file_intake: intake, employee_ssn: spouse_ssn_from_fixture, box14_ui_wf_swf: 179.78) }
+        include_examples "does not display W2 warnings"
+      end
+
+      context "when two or more w2s and box14_ui_wf_swf and fli are valid on both" do
+        let!(:state_file_w2_1) { create(:state_file_w2, state_file_intake: intake, box14_ui_wf_swf: 179.78, box14_fli: 145.26) }
+        let!(:state_file_w2_2) { create(:state_file_w2, state_file_intake: intake, box14_ui_wf_swf: 179.78, box14_fli: 145.26) }
+        include_examples "does not display W2 warnings"
+      end
+    end
+
+    context "when W2 box 14 warnings are present" do
+      let(:intake) { create(:state_file_nj_intake) }
+
+      shared_examples "displays at least one W2 warning" do
+        it "displays at least one W2 warning" do
+          get :edit, params: params
+          expect(response.body).to have_text "We need to double-check some information"
+        end
+      end
```

Comment:
[dust/pebble] should we test for the correct number of W2 warnings? "at least one" feels imprecise.

I think you can do this by passing a `count` to the matcher

Reply:
ooh i like that. I edited it to expect exactly one warning. thanks!